src/backend/distributed/shared_library_init.c

```diff
@@ -923,7 +936,7 @@ RegisterCitusConfigVariables(void)
 					 "queries that touch thousands of shards and/or that involve table "
 					 "repartitioning."),
 		&TaskExecutorType,
-		MULTI_EXECUTOR_REAL_TIME,
+		MULTI_ADAPTIVE_EXECUTOR,
```

Comment:
do we still need to support pool size 0 if we you can disable the executor here?

Reply:
Wouldn't MaxPoolSize == 0 become obsolete at the same time or sooner?

Reply:
I think there is a minor misunderstanding here.

You cannot disable adaptive executor with setting the taskExecutor to realtime with the current code. The only way is to set the pool size. This is just to make things to look a bit more compatible.

Do you think should we disable when this GUC is not set to adaptive?

Reply:
I'd have a slight preference for that, but I don't feel so strongly about it that I want to delay things.